File: src/main/scala/com/github/lancelet/csvee/CsvError.scala
package com.github.lancelet.csvee

sealed trait CsvError

case object CsvErrorFreeQuoteInField        extends CsvError
case object CsvErrorExpectedSeparator       extends CsvError
case object CsvErrorSeparatorTerminatedLine extends CsvError
case object CsvErrorMissingQuote            extends CsvError

File: src/main/java/com/github/lancelet/csvee/CsvError.java
package com.github.lancelet.csvee;

public final class CsvError {
    public CsvError(Type type, int column) {
        this.type   = type;
        this.column = column;
    }

    public final Type type;
    public final int column;

    public static enum Type {
        ClosingQuoteMissing,
        FreeQuoteInUnquotedField,
        FieldNotFollowedBySeparator,
        LineTerminatedBySeparator
    }
}
